share/spack/docker/package-index/README.rst

```diff
@@ -0,0 +1,45 @@
+============================================
+The packages.spack.io Package Index REST API
+============================================
+
+This directory provides the docker recipe for the Spack package index on https://packages.spack.io
+
+On each merge to ``develop``, DockerHub builds a new image ``spack/packages.spack.io`` which is configured in:
+  https://cloud.docker.com/u/spack/repository/docker/spack/packages.spack.io/builds/edit
+
+------------
+The REST API
+------------
+
+The API is a simple, file-based JSON index.
+A specific package can be queried via the URI syntax:
+``https://packages.spack.io/api/:firstLetter/:packageName.json``
```

Comment:
This is more curiosity on my part than anything else, but why the `:firstLetter` part?  Is that part of some standard that this service is trying to adhere to?

Reply:
This is just me trying to implement a very naïve load balancing to split the cost of an 5k inode `ls`.
```
ls var/spack/repos/builtin/packages/ | wc -l
3695
```

Reply:
I am anticipating 100k+ packages in Spack